profiler/src/ProfilerEngine/Datadog.Profiler.Native.Windows/ETW/IpcServer.cpp

```diff
@@ -114,12 +114,13 @@ std::unique_ptr<IpcServer> IpcServer::StartAsync(
         return nullptr;
     }
 
-    auto server = std::make_unique<IpcServer>(
+    // the lifetime of this instance is the lifetime of the application (i.e. it won't be deleted to avoid random crashes)
+    auto server = new IpcServer(
```

Comment:
Do we have something with an equivalent lifetime where we could register this ? Or would we need a sync mechanism to request a shutdown ?

Reply:
In the current implementation, the instance is destroyed AFTER the Shutdown is received from the CLR and we see a crash. Our guess is that the other native thread is then trying to access it